### PR TITLE
Adjust TeamCity build pipeline

### PR DESCRIPTION
**Pull request by donat**
This PR contains the following changes:
- remove unused `Gradle_AgentTest` pipeline (was already removed with a patch)
- use an explicit integer as ID for `TestCoverage`
- extract min/max JVM version to enum

### CODE REVIEW

File: .teamcity/Gradle_Check/model/CIBuildModel.kt [hunk 1]
```diff
@@ -1,10 +1,6 @@
 package model
 
-import common.BuildCache
-import common.JvmVendor
-import common.JvmVersion
-import common.Os
-import common.builtInRemoteBuildCacheNode
+import common.*
```
Comment (wolfs) [inline .teamcity/Gradle_Check/model/CIBuildModel.kt]: Can we not use `*` imports?

File: .teamcity/Gradle_Check_dsl.iml [hunk 2]
```diff
@@ -1,102 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (wolfs) [inline .teamcity/Gradle_Check_dsl.iml]: I remember keeping the `.iml` file intentionally under version control. Is that not necessary any more? Shall we add this file to `.gitignore` if it isn't already added?

File: .teamcity/Gradle_Check/model/CIBuildModel.kt (hunk 1 above)
Comment (donat) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: Sorry, it was done by IDEA.

Comment (donat) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: Fixed.

File: .teamcity/Gradle_Check_dsl.iml (hunk 2 above)
Comment (donat) [reply .teamcity/Gradle_Check_dsl.iml]: There might be something wrong with my setup. I removed this accidental change.

**Comment by donat** [comment]
Merged manually without raising the maximum Java version.

**Comment by donat** [comment]
FYI I've cherry-picked the change to release to keep the build IDs consistent, in case we do a 5.3.1 release.

**Pull request merged**